Geometry/HcalAlgo/plugins/dd4hep/DDHCalEndcapAlgo.cc

```diff
@@ -0,0 +1,979 @@
+  double riDip;                        //Inner radius at the dip point
+  double roDip;                        //Outer radius at the dip point
+  double heboxDepth;                   //Depth of the HE box
+  double drEnd;                        //Shift in R for the end absorber
+  double angTop;                       //Angle of top end of HE
+  double angBot;                       //Angle of the bottom end of HE
+  double angGap;                       //Gap angle (in degrees)
+  double slope;                        //Slope of the gap on HE side
+  std::string absMat;                  //Absorber     material
+  int modules;                         //Number of modules
+  std::vector<std::string> modName;    //Name
+  std::vector<std::string> modMat;     //Material
+  std::vector<int> modType;            //Type (0/1 for front/standard)
+  std::vector<int> sectionModule;      //Number of sections in a module
+  std::vector<int> layerN;             //Number of layers
+  std::vector<int> layerN0;            //Layer numbers in section 0
+  std::vector<int> layerN1;            //Layer numbers in section 1
+  std::vector<int> layerN2;            //Layer numbers in section 2
+  std::vector<int> layerN3;            //Layer numbers in section 3
+  std::vector<int> layerN4;            //Layer numbers in section 4
+  std::vector<int> layerN5;            //Layer numbers in section 5
+  std::vector<double> thick;           //Thickness of absorber/air
+  std::vector<double> trimLeft;        //Trimming of left  layers in module
+  std::vector<double> trimRight;       //Trimming of right layers in module
+  std::vector<double> zminBlock;       //Minimum Z
+  std::vector<double> zmaxBlock;       //Maximum Z
+  std::vector<double> rinBlock1;       //Inner Radius
+  std::vector<double> routBlock1;      //Outer Radius at zmin
+  std::vector<double> rinBlock2;       //Inner Radius
+  std::vector<double> routBlock2;      //Outer Radius at zmax
+  int phiSections;                     //Number of phi sections
+  std::vector<std::string> phiName;    //Name of Phi sections
+  int layers;                          //Number of layers
+  std::vector<std::string> layerName;  //Layer Names
+  std::vector<int> layerType;          //Detector type in each layer
+  std::vector<double> layerT;          //Layer thickness (plastic + scint.)
+  std::vector<double> scintT;          //Scintillator thickness
+  std::string plastMat;                //Plastic      material
+  std::string scintMat;                //Scintillator material
+  std::string rotmat;                  //Rotation matrix for positioning
+  std::string idName;                  //Name of the "parent" volume.
+  std::string idNameSpace;             //Namespace of this and ALL sub-parts
+  int idOffset;                        // Geant4 ID's...    = 4000;
+  double tolPos, tolAbs;               //Tolerances
+
+  HCalEndcapAlgo() { throw cms::Exception("HCalGeom") << "Wrong initialization to HcalEndcapAlgo"; }
```

Comment:
@bsunanda @kpedro88 should not we apply the same argument as in https://github.com/cms-sw/cmssw/pull/28068#pullrequestreview-293333187 and set this to a delete constructor?

Reply:
Yes, please do

Reply:
Done